### PR TITLE
Remove most imports of UIKit

### PR DESCRIPTION
**Pull request by twizmwazin**
# Checklist
- [x] I have read [CONTRIBUTING.md](./CONTRIBUTING.md)
- [x] I have described what this PR contains
- [ ] This PR addresses one or more open issues that were assigned to me:
      - *list issue(s) here*
- [x] If this PR alters the UI, I have attached pictures/videos

# Pull Request Information

## About this Pull Request
**Describe what this PR contains in as much detail as possible**
This PR remove almost all of the references to UIKit. Most of these were nothing more than deleting the import, the photo saving required a change. I would appreciate if someone could check I did that correctly!

## Screenshots and Videos
**In case this PR changes something in the UI, please include screenshots or videos of this new feature**
None, any functional change is a bug

## Additional Context
**Any additional context you'd like to add to help us review this PR**
Removing UIKit dependency and being 100% on SwiftUI would allow native Mac support someday.

### CODE REVIEW

File: Mlem/Temp Image Viewer/ImageSaver.swift [hunk 1]
```diff
@@ -7,16 +7,19 @@
...
-    @objc func saveCompleted(_ image: UIImage, didFinishSavingWithError error: Error?, contextInfo: UnsafeRawPointer) {
-        print("Save finished!")
+        PHPhotoLibrary.shared().performChanges({
+            let creationRequest = PHAssetCreationRequest.forAsset()
+            creationRequest.addResource(with: .photo, data: imageData, options: nil)
+        }, completionHandler: { success, error in
+            if success {
+                print("Save finished!")
+            } else {
+                print("Error saving photo: \(String(describing: error?.localizedDescription))")
```
Comment (WestonHanners) [inline Mlem/Temp Image Viewer/ImageSaver.swift:21]: Probably good to call out we should maybe handle this better in the future.

File: Mlem/Notifications/NotificationDisplayer.swift [hunk 2]
```diff
@@ -7,10 +7,9 @@
 //
 
 import SwiftUI
-import UIKit
 
 /// A class responsible for displaying important notifications to the user
```
Comment (WestonHanners) [inline Mlem/Notifications/NotificationDisplayer.swift:11]: why is this an enum, this makes it a value type instead of a reference type.

**Review by EricBAndrews — APPROVED** [review]
Looks good to me--image saving works as intended, and the unused import removal should be guaranteed safe by the compiler. I'm going to hold off on merging until a team member with more significant UI experience has had a chance to look through, though.

**Comment by EricBAndrews** [comment]
Should be good to merge with a rebase + passing CI

**Comment by twizmwazin** [comment]
As far as I can tell the CI is failing due to my fork not having write permission on some resource. The tests themselves appear to be passing.

**Pull request merged**